realtime messages_test fail

I'm new to Go. The way I would fix the following tests would be:
```
messageChannel := make(chan models.Message)
err := c.SubscribeToMessageStream(&general, messageChannel)
```
But it might be better to return a channel (instead of editing the test)?
```
./messages_test.go:17:22: cannot assign 1 values to 2 variables
./messages_test.go:17:51: not enough arguments in call to c.SubscribeToMessageStream
	have (*models.Channel)
	want (*models.Channel, chan models.Message)
./messages_test.go:52:22: cannot assign 1 values to 2 variables
./messages_test.go:52:51: not enough arguments in call to c.SubscribeToMessageStream
	have (*models.Channel)
	want (*models.Channel, chan models.Message)
```

--- a/models/user.go
+++ b/models/user.go
@@ -1,6 +1,8 @@
 package models
 
 type User struct {
-	Id       string `json:"_id"`
-	UserName string `json:"username"`
+	Id           string `json:"_id"`
+	UserName     string `json:"username"`
+	Token        string `json:"token"`
+	TokenExpires int64  `json:"tokenExpires"`
 }

--- a/models/userCredentials.go
+++ b/models/userCredentials.go
@@ -1,6 +1,9 @@
 package models
 
 type UserCredentials struct {
+	ID    string `json:"id"`
+	Token string `json:"token"`
+
 	Email    string `json:"email"`
 	Name     string `json:"name"`
 	Password string `json:"pass"`

--- a/realtime/client_test.go
+++ b/realtime/client_test.go
@@ -1,6 +1,7 @@
 package realtime
 
 import (
+	"net/url"
 	"testing"
 
 	"github.com/RocketChat/Rocket.Chat.Go.SDK/common_testing"
@@ -15,10 +16,10 @@ var (
 func getLoggedInClient(t *testing.T) *Client {
 
 	if client == nil {
-		c, err := NewClient(common_testing.Host, common_testing.Port, true)
+		c, err := NewClient(&url.URL{Host: common_testing.Host + ":" + common_testing.Port}, true)
 		assert.Nil(t, err, "Couldn't create realtime client")
 
-		err = c.RegisterUser(&models.UserCredentials{
+		_, err = c.RegisterUser(&models.UserCredentials{
 			Email:    common_testing.GetRandomEmail(),
 			Name:     common_testing.GetRandomString(),
 			Password: common_testing.GetRandomString()})

--- a/realtime/messages_test.go
+++ b/realtime/messages_test.go
@@ -13,15 +13,18 @@ func TestClient_SubscribeToMessageStream(t *testing.T) {
 
 	general := models.Channel{Id: "GENERAL"}
 	textToSend := "RealtimeTest"
+	messageChannel := make(chan models.Message, 1)
 
-	messageChannel, err := c.SubscribeToMessageStream(&general)
+	err := c.SubscribeToMessageStream(&general, messageChannel)
 
 	assert.Nil(t, err, "Function returned error")
 	assert.NotNil(t, messageChannel, "Function didn't returned general")
 
-	sendAndAssertNoError(t, c, &general, textToSend)
-	sendAndAssertNoError(t, c, &general, textToSend)
-	sendAndAssertNoError(t, c, &general, textToSend)
+	go func() {
+		sendAndAssertNoError(t, c, &general, textToSend)
+		sendAndAssertNoError(t, c, &general, textToSend)
+		sendAndAssertNoError(t, c, &general, textToSend)
+	}()
 
 	receivedMessage1 := <-messageChannel
 	receivedMessage2 := <-messageChannel
@@ -31,6 +34,7 @@ func TestClient_SubscribeToMessageStream(t *testing.T) {
 	assertMessage(t, receivedMessage2)
 	assertMessage(t, receivedMessage3)
 }
+
 func assertMessage(t *testing.T, message models.Message) {
 	assert.NotNil(t, message.Id, "Id was not set")
 	assert.Equal(t, "GENERAL", message.ChannelId, "Wrong channel id")
@@ -38,6 +42,7 @@ func assertMessage(t *testing.T, message models.Message) {
 	assert.NotNil(t, message.User.Id, "UserId was not set")
 	assert.NotNil(t, message.User.UserName, "Username was not set")
 }
+
 func sendAndAssertNoError(t *testing.T, c *Client, channel *models.Channel, text string) {
 	m, err := c.SendMessage(channel, text)
 	assert.Nil(t, err, "Error while sending message")
@@ -48,9 +53,9 @@ func TestClient_SubscribeToMessageStream_UnknownChannel(t *testing.T) {
 
 	c := getLoggedInClient(t)
 	channel := models.Channel{Id: "unknown"}
+	messageChannel := make(chan models.Message, 1)
 
-	messageChannel, err := c.SubscribeToMessageStream(&channel)
+	err := c.SubscribeToMessageStream(&channel, messageChannel)
 
 	assert.NotNil(t, err, "Function didn't return error")
-	assert.Nil(t, messageChannel, "Function returned channel, but shouldn't")
 }

--- a/realtime/users.go
+++ b/realtime/users.go
@@ -47,27 +47,38 @@ func (c *Client) RegisterUser(credentials *models.UserCredentials) (*models.User
 	return user, nil
 }
 
-// Login a user. The password and the email are not allowed to be nil.
+// Login a user.
+// token shouldn't be nil, otherwise the password and the email are not allowed to be nil.
 //
 // https://rocket.chat/docs/developer-guides/realtime-api/method-calls/login/
 func (c *Client) Login(credentials *models.UserCredentials) (*models.User, error) {
+	var request interface{}
+	if credentials.Token != "" {
+		request = ddpTokenLoginRequest{
+			Token: credentials.Token,
+		}
+	} else {
+		digest := sha256.Sum256([]byte(credentials.Password))
+		request = ddpLoginRequest{
+			User: ddpUser{Email: credentials.Email},
+			Password: ddpPassword{
+				Digest:    hex.EncodeToString(digest[:]),
+				Algorithm: "sha-256",
+			},
+		}
+	}
 
-	digest := sha256.Sum256([]byte(credentials.Password))
+	rawResponse, err := c.ddp.Call("login", request)
+	if err != nil {
+		return nil, err
+	}
 
-	rawResponse, err := c.ddp.Call("login", ddpLoginRequest{
-		User:     ddpUser{Email: credentials.Email},
-		Password: ddpPassword{Digest: hex.EncodeToString(digest[:]), Algorithm: "sha-256"}})
+	user := getUserFromData(rawResponse.(map[string]interface{}))
+	if credentials.Token == "" {
+		credentials.ID, credentials.Token = user.Id, user.Token
+	}
 
-	return getUserFromData(rawResponse.(map[string]interface{})), err
-}
-
-// Auth a user. Using an authentication token to login.
-//
-// https://rocket.chat/docs/developer-guides/realtime-api/method-calls/login/#using-an-authentication-token
-func (c *Client) Auth(token string) (*models.User, error) {
-	rawResponse, err := c.ddp.Call("login", ddpTokenLoginRequest{Token: token})
-
-	return getUserFromData(rawResponse.(map[string]interface{})), err
+	return user, nil
 }
 
 func getUserFromData(data interface{}) *models.User {

--- a/realtime/users.go
+++ b/realtime/users.go
@@ -3,6 +3,7 @@ package realtime
 import (
 	"crypto/sha256"
 	"encoding/hex"
+	"strconv"
 
 	"github.com/Jeffail/gabs"
 	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
@@ -60,11 +61,23 @@ func (c *Client) Login(credentials *models.UserCredentials) (*models.User, error
 	return getUserFromData(rawResponse.(map[string]interface{})), err
 }
 
+// Auth a user. Using an authentication token to login.
+//
+// https://rocket.chat/docs/developer-guides/realtime-api/method-calls/login/#using-an-authentication-token
+func (c *Client) Auth(token string) (*models.User, error) {
+	rawResponse, err := c.ddp.Call("login", ddpTokenLoginRequest{Token: token})
+
+	return getUserFromData(rawResponse.(map[string]interface{})), err
+}
+
 func getUserFromData(data interface{}) *models.User {
 	document, _ := gabs.Consume(data)
 
+	expires, _ := strconv.ParseFloat(stringOrZero(document.Path("tokenExpires.$date").Data()), 64)
 	return &models.User{
-		Id: stringOrZero(document.Path("id").Data()),
+		Id:           stringOrZero(document.Path("id").Data()),
+		Token:        stringOrZero(document.Path("token").Data()),
+		TokenExpires: int64(expires),
 	}
 }
 

--- a/rest/channels_test.go
+++ b/rest/channels_test.go
@@ -2,6 +2,7 @@ package rest
 
 import (
 	"testing"
+
 	"github.com/stretchr/testify/assert"
 )
 
@@ -62,8 +63,12 @@ func TestRocket_GetChannelInfo(t *testing.T) {
 
 	assert.Equal(t, general.Id, updatedChannelInfo.Id)
 	assert.NotEmpty(t, updatedChannelInfo.Name)
-	assert.NotEmpty(t, updatedChannelInfo.T)
+	assert.NotEmpty(t, updatedChannelInfo.Type)
 	assert.NotEmpty(t, updatedChannelInfo.UpdatedAt)
 	assert.NotEmpty(t, updatedChannelInfo.Timestamp)
-	assert.NotZero(t, len(updatedChannelInfo.UserNames))
+	// API has changed
+	// Please use the `channel/dm/group.members` endpoint. This is disabled for performance reasons
+	// https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-api/server/v1/channels.js#L424
+	// https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-api/server/api.js#L15
+	// assert.NotZero(t, len(updatedChannelInfo.UserNames))
 }

--- a/rest/client_test.go
+++ b/rest/client_test.go
@@ -1,7 +1,7 @@
 package rest
 
 import (
-	"fmt"
+	"net/url"
 	"testing"
 
 	"github.com/RocketChat/Rocket.Chat.Go.SDK/common_testing"
@@ -32,7 +32,7 @@ func getAuthenticatedClient(t *testing.T, name, email, password string) *Client 
 	client := Client{Protocol: common_testing.Protocol, Host: common_testing.Host, Port: common_testing.Port}
 	credentials := models.UserCredentials{Name: name, Email: email, Password: password}
 
-	rtClient, err := realtime.NewClient(fmt.Sprintf("%s:%s", common_testing.Host, common_testing.Port), false, true)
+	rtClient, err := realtime.NewClient(&url.URL{Host: common_testing.Host + ":" + common_testing.Port}, true)
 	assert.Nil(t, err)
 	_, regErr := rtClient.RegisterUser(&credentials)
 	assert.Nil(t, regErr)

--- a/rest/information_test.go
+++ b/rest/information_test.go
@@ -10,33 +10,12 @@ import (
 func TestRocket_GetServerInfo(t *testing.T) {
 	rocket := Client{Protocol: common_testing.Protocol, Host: common_testing.Host, Port: common_testing.Port}
 
+	// API has changed
+	// https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-api/server/v1/misc.js#L13
 	info, err := rocket.GetServerInfo()
 
 	assert.Nil(t, err)
 	assert.NotNil(t, info)
 
 	assert.NotEmpty(t, info.Version)
-
-	assert.NotEmpty(t, info.Build.Arch)
-	assert.NotZero(t, info.Build.CpuCount)
-	assert.NotEmpty(t, info.Build.Platform)
-	assert.NotEmpty(t, info.Build.Date)
-	assert.NotZero(t, info.Build.FreeMemory)
-	assert.NotZero(t, info.Build.TotalMemory)
-	assert.NotEmpty(t, info.Build.NodeVersion)
-	assert.NotEmpty(t, info.Build.OsRelease)
-
-	assert.NotEmpty(t, info.Travis.Branch)
-	assert.NotEmpty(t, info.Travis.BuildNumber)
-	assert.NotEmpty(t, info.Travis.Tag)
-
-	assert.NotEmpty(t, info.Commit.Author)
-	assert.NotEmpty(t, info.Commit.Branch)
-	assert.NotEmpty(t, info.Commit.Date)
-	assert.NotEmpty(t, info.Commit.Hash)
-	assert.NotEmpty(t, info.Commit.Subject)
-	assert.NotEmpty(t, info.Commit.Tag)
-
-	assert.NotEmpty(t, info.ImageMagick.Version)
-	assert.NotNil(t, info.GraphicsMagick)
 }

